### DISCUSSION TITLE
Is there a way to use this in CI

### DISCUSSION THREAD

**Discussion by zyga** [streetsidesoftware/vscode-spell-checker#1063] [Q&A] [answered]
I would like to use vscode-spell-checker, specifically in conjunction with the repository-specific local data, to check commits for spelling mistakes in CI. Has anyone done something like that?

**Comment by Jason3S** [comment, marked as answer]
@zyga,

This spell checker uses the [cspell](https://github.com/streetsidesoftware/cspell/tree/main/packages/cspell) library to do the spell checking.

I suggest moving your settings to a `cspell.json` file so the cspell command line tool can pick them up.

Example to check all `*.ts` files:
```sh
npx cspell "**/*.ts"
```

**Comment by mrienstra** [comment]
There is also an official GitHub Action:
* Repo: https://github.com/streetsidesoftware/cspell-action
* Marketplace: https://github.com/marketplace/actions/cspell-action
